crates/kitsune_p2p/proxy/src/tx2.rs

```diff
@@ -278,7 +321,7 @@ impl ProxyEpHnd {
         sub_ep_hnd: EpHnd,
         logic_hnd: LogicChanHandle<EpEvent>,
         backoff: Backoff,
-        client_of_remote_proxy: Option<ProxyUrl>,
+        cur_proxy_url: Arc<parking_lot::RwLock<Option<ProxyUrl>>>,
```

Comment:
Maybe we can use `Share` so we don't get any weird `.read()` deadlocks that seem to be common when using naked locks?

Reply:
:+1: done